### ISSUE TITLE
[ITensorGaussianMPS] [BUG] Bug in slater determinant to MPS conversion

### ISSUE THREAD

**Issue by srwhite59** [ITensor/ITensors.jl#1041]
**Description of bug**

slater to mps fails, dies with error message

**Minimal code demonstrating the bug or unexpected behavior**

If applicable, provide a minimal code that can be run to demonstrate the bug or unexpected behavior.

If you are unable to construct a minimal code that demonstrates the bug or unexpected behavior, provide detailed steps for how to reproduce the behavior you are seeing.

<details><summary>Minimal runnable code</summary><p>

```julia
#!/opt/apps/julia/1.8.2/bin/julia
using ITensors, LinearAlgebra
using ITensorGaussianMPS

N=803
Nup=Ndn=4

sites = siteinds("Electron",N; conserve_qns=true)
psiup,~,~ = svd(rand(N,Nup))
psidn,~,~ = svd(rand(N,Ndn))
psi0 = slater_determinant_to_mps(sites, psiup, psidn;eigval_cutoff = 1e-6,
                                         cutoff = 1e-9, maxdim  = 50)

```
</p></details>

**Expected output or behavior**

Produce an MPS without error.

**Actual output or behavior**

<details><summary>Output of minimal runnable code</summary><p>

```julia
ERROR: LoadError: MethodError: no method matching block(::Index{Vector{Pair{QN, Int64}}}, ::UInt64)
Closest candidates are:
  block(::Vector{Int64}, ::Integer) at ~/.julia/packages/NDTensors/FHyfI/src/blocksparse/blockdims.jl:205
Stacktrace:
  [1] svd(T::NDTensors.BlockSparseTensor{Float64, 2, Tuple{Index{Vector{Pair{QN, Int64}}}, Index{Vector{Pair{QN, Int64}}}}, NDTensors.BlockSparse{Float64, Vector{Float64}, 2}}; kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:tags, :alg), Tuple{TagSet, String}}})
    @ NDTensors ~/.julia/packages/NDTensors/FHyfI/src/blocksparse/linearalgebra.jl:115
  [2] svd(A::ITensor, Linds::Vector{Index{Vector{Pair{QN, Int64}}}}; kwargs::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:tags, :alg), Tuple{TagSet, String}}})
    @ ITensors ~/.julia/packages/ITensors/5sSxp/src/decomp.jl:109
  [3] factorize_svd(A::ITensor, Linds::Vector{Index{Vector{Pair{QN, Int64}}}}; kwargs::Base.Pairs{Symbol, TagSet, Tuple{Symbol}, NamedTuple{(:tags,), Tuple{TagSet}}})
    @ ITensors ~/.julia/packages/ITensors/5sSxp/src/decomp.jl:378
  [4] factorize(A::ITensor, Linds::Vector{Index{Vector{Pair{QN, Int64}}}}; kwargs::Base.Pairs{Symbol, TagSet, Tuple{Symbol}, NamedTuple{(:tags,), Tuple{TagSet}}})
    @ ITensors ~/.julia/packages/ITensors/5sSxp/src/decomp.jl:498
  [5] orthogonalize!(M::MPS, j::Int64; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ ITensors ~/.julia/packages/ITensors/5sSxp/src/mps/abstractmps.jl:1363
  [6] orthogonalize!
    @ ~/.julia/packages/ITensors/5sSxp/src/mps/abstractmps.jl:1352 [inlined]
  [7] #orthogonalize#664
    @ ~/.julia/packages/ITensors/5sSxp/src/mps/abstractmps.jl:1398 [inlined]
  [8] orthogonalize
    @ ~/.julia/packages/ITensors/5sSxp/src/mps/abstractmps.jl:1396 [inlined]
  [9] product(o::ITensor, ψ::MPS, ns::Vector{Int64}; move_sites_back::Bool, apply_dag::Bool, kwargs::Base.Pairs{Symbol, Real, Tuple{Symbol, Symbol}, NamedTuple{(:cutoff, :maxdim), Tuple{Float64, Int64}}})
    @ ITensors ~/.julia/packages/ITensors/5sSxp/src/mps/abstractmps.jl:1844
 [10] product(As::Vector{ITensor}, ψ::MPS; move_sites_back::Bool, kwargs::Base.Pairs{Symbol, Real, Tuple{Symbol, Symbol}, NamedTuple{(:cutoff, :maxdim), Tuple{Float64, Int64}}})
    @ ITensors ~/.julia/packages/ITensors/5sSxp/src/mps/abstractmps.jl:1959
 [11] #MPS#7
    @ ~/.julia/packages/ITensorGaussianMPS/5lyPk/src/gmps.jl:248 [inlined]
 [12] correlation_matrix_to_mps(s::Vector{Index{Vector{Pair{QN, Int64}}}}, Λ_up::Matrix{Float64}, Λ_dn::Matrix{Float64}; eigval_cutoff::Float64, maxblocksize::Int64, kwargs::Base.Pairs{Symbol, Real, Tuple{Symbol, Symbol}, NamedTuple{(:cutoff, :maxdim), Tuple{Float64, Int64}}})
    @ ITensorGaussianMPS ~/.julia/packages/ITensorGaussianMPS/5lyPk/src/gmps.jl:419
 [13] slater_determinant_to_mps(s::Vector{Index{Vector{Pair{QN, Int64}}}}, Φ_up::Matrix{Float64}, Φ_dn::Matrix{Float64}; kwargs::Base.Pairs{Symbol, Real, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:eigval_cutoff, :cutoff, :maxdim), Tuple{Float64, Float64, Int64}}})
    @ ITensorGaussianMPS ~/.julia/packages/ITensorGaussianMPS/5lyPk/src/gmps.jl:321
 [14] top-level scope
    @ ~/bin/te.jl:11
in expression starting at /data/homezvol1/srwhite/bin/te.jl:11

```
</p></details>

**Version information**

 - Output from `versioninfo()`:
```julia
julia> versioninfo()
Julia Version 1.8.2
Commit 36034abf260 (2022-09-29 15:21 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 40 × Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, cascadelake)
  Threads: 1 on 40 virtual cores
Environment:
  LD_LIBRARY_PATH = /opt/intel/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64:/opt/intel/compilers_and_libraries_2019.5.281/linux/mkl/lib/intel64:/opt/intel/compilers_and_libraries_2019.5.281/linux/tbb/lib/intel64:/opt/apps/julia/1.8.2/lib:/opt/intel/compilers_and_libraries_2019.5.281/linux/mkl/lib/intel64_lin:/opt/intel/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64_lin
  julia__LDFLAGS = -L/opt/apps/julia/1.8.2/lib -Wl,-rpath,/opt/apps/julia/1.8.2/lib
  julia__PREFIX = /opt/apps/julia/1.8.2
  LD_LIBRARY_PATH_modshare = /opt/intel/compilers_and_libraries_2019.5.281/linux/mkl/lib/intel64:1:/opt/intel/compilers_and_libraries_2019.5.281/linux/tbb/lib/intel64:1:/opt/apps/julia/1.8.2/lib:1:/opt/intel/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64_lin:1:/opt/intel/compilers_and_libraries_2019.5.281/linux/mkl/lib/intel64_lin:1:/opt/intel/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64:1
  julia__CPPFLAGS = -I/opt/apps/julia/1.8.2/include
  LD_RUN_PATH = /opt/intel/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64:/opt/intel/compilers_and_libraries_2019.5.281/linux/mkl/lib/intel64:/opt/intel/compilers_and_libraries_2019.5.281/linux/tbb/lib/intel64
  LD_RUN_PATH_modshare = /opt/intel/compilers_and_libraries_2019.5.281/linux/mkl/lib/intel64:1:/opt/intel/compilers_and_libraries_2019.5.281/linux/tbb/lib/intel64:1:/opt/intel/compilers_and_libraries_2019.5.281/linux/compiler/lib/intel64:1

```
 - Output from `using Pkg; Pkg.status("ITensors")`:
```julia
julia> using Pkg; Pkg.status("ITensors")
Status `~/.julia/environments/v1.8/Project.toml`
⌅ [9136182c] ITensors v0.2.16
Info Packages marked with ⌅ have new versions available but compatibility constraints restrict them from upgrading. To see why use `status --outdated`

```

**Comment by mtfishman** [comment]
@srwhite59 the code you posted above runs fine for me with ITensors v0.3.23. It looks like the issue is that something is blocking you from upgrading to the latest version of ITensors.jl, try running `status --outdated` in Pkg mode to see what is blocking you from upgrading.

**Comment by srwhite59** [comment]
I tried to include this in the bug report originally--somehow I didn't get it in.
(@v1.8) pkg> status --outdated
Status `~/.julia/environments/v1.8/Project.toml`
  [2be41995] ITensorGaussianMPS v0.0.1 `https://github.com/ITensor/ITensorGaussianMPS.jl#main` (<v0.0.4)
⌅ [9136182c] ITensors v0.2.16 (<v0.3.23): ITensorGaussianMPS

**Comment by mtfishman** [comment]
It looks like you are using a local version of `ITensorGaussianMPS` that is fixed to v0.0.1. Maybe try `] free ITensorGaussianMPS` and then updating your packages to get the latest version.

**Comment by mtfishman** [comment]
@srwhite59 I noticed the installation instructions for the `ITensorGaussianMPS` package were outdated (it said the package wasn't registered but now it is), which is probably why your local setup didn't have the latest version. I've updated the instructions: https://github.com/ITensor/ITensors.jl/blob/main/ITensorGaussianMPS/README.md#installation
